pod-analyzer/src/FailedPage.tsx

```diff
@@ -1,90 +1,151 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 
-const FailedPage = () => (
-    <div 
-        style={{ 
-            display: 'flex', 
-            justifyContent: 'center', 
-            alignItems: 'center', 
-            flexDirection: 'column', 
-            width: '700%', 
-            padding: '24px' 
-        }}
-    >
-        <hr 
-            style={{ 
-                width: '100%', 
-                border: 'none', 
-                borderTop: '2px dashed black', 
-                marginBottom: '36px' 
-            }} 
-        />
-        <h2>More Info About Failed Pods</h2>
+interface Pod {
+    name: string;
+    nodeName: string;
+    podIP: string;
+    updateTime: string;
+}
+
+const FailedPage = () => {
+    const [failedPods, setFailedPods] = useState<Pod[]>([]);
+
+    useEffect(() => {
+        const fetchFailedPods = async () => {
+            try {
+                const response = await fetch('/api/v1/pods');
```

Comment:
[suggestion] we should avoid loading data from time to time on the same page, especially these data could be shared through the components via props or context